azerothcore/azerothcore-wotlk
# fix(Core/spell) Add an intervene condition to avoid stealth removal

##### CHANGES PROPOSED:

This PR fixes the problem of warrior's Intervene removing stealth mechanics. It avoids the wrong removal of stealth on rogues, prowl of druids, racial of night elfs and invisibility of mages due to intervene trigger (spell id 59667)

##### ISSUES ADDRESSED:

    Closes issue #2472

##### TESTS PERFORMED:

    Tested build on Ubuntu 18.04.2 LTS x64 / cmake 3.10.2 / clang 6.0.0 / MySQL 5.7.28
    Tested successfully in-game

##### HOW TO TEST THE CHANGES:

Without this PR:
    Create 2 groups for arena skirmish 2v2, one with a warrior with intervene spell, and other with a rogue with stealth.
    Due to mage invisibility having a diferent removal on the code, I highly recommend making the arena teams this way: 
      -Team 1: Night Elf Warrior - rogue
      -Team 2; Warrior - mage
    This way you can test diferent conditions for the same fix.
    Call a .skirmish on any arena, and use stealth with the rogue, mage or night elf, away from the warrior, but in a specific spot of arena, to see how it shows once stealth/invisibility is removed.
    Use intervene with the warrior on your teammate, and see the enemy appear.

Without the changes in L2955, the fix will be incomplete due to intervene still removing mage's invisibility.


##### KNOWN ISSUES AND TODO LIST:
Check if is possible to work with intervene's trigger flags on database to avoid the removal of stealth mechanics.

##### Target branch(es):


 
## How to test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR
